napari/components/_viewer_key_bindings.py

```diff
@@ -150,8 +150,8 @@ def hold_for_pan_zoom(viewer: ViewerModel):
         yield
         return
     previous_mode = selected_layer.mode
-    if previous_mode != selected_layer._modeclass.PAN_ZOOM:
-        selected_layer.mode = selected_layer._modeclass.PAN_ZOOM
+    if previous_mode != (pan_zoom := selected_layer._modeclass.PAN_ZOOM):  # type: ignore[attr-defined]
+        selected_layer.mode = pan_zoom
```

Comment:
Nit: I think the walrus operator hurts readability here over the previous form because the reader has to unpack the two expressions in the condition now. I guess you need two ignores in that case though, so I don't care too much.

More generally, typing of `Mode` feels a bit broken and this is just one symptom. Do we have an issue to track a longer term fix for that?

Reply:
I agree. This walrus operator is obsolete as this part of the code does not contain any heavy calculations that need to be cached.

Reply:
I unpacked the walrus operator and added a comment to explain why we need the `type: ignore`.

We do have an issue to track the more general problem with layer modes: https://github.com/napari/napari/issues/6005